public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -166,11 +168,12 @@ export const RolePickerMenu = ({
 
   const onUpdateInternal = () => {
     const selectedCustomRoles: string[] = [];
+    // TODO: needed?
     for (const key in selectedOptions) {
       const roleUID = selectedOptions[key]?.uid;
       selectedCustomRoles.push(roleUID);
     }
-    onUpdate(selectedCustomRoles, selectedBuiltInRole);
+    onUpdate(selectedOptions, selectedBuiltInRole);
```

Comment:
I guess we need the UIDs for this to work?

if not; we could remove selectedCustomRoles entirely